NewApp/ngsa-csharp/Ngsa.App/Controllers/DataService.cs

```diff
@@ -46,14 +47,16 @@ public static async Task<IActionResult> Read<T>(string path, string queryString
 
             string fullPath = path.Trim();
 
-            if (!string.IsNullOrWhiteSpace(queryString))
-            {
-                fullPath += $"?{queryString.Trim()}";
-            }
-
             try
             {
-                HttpResponseMessage resp = await Client.GetAsync(fullPath);
+                HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Get, fullPath);
+
+                if (cVector != null)
+                {
+                    req.Headers.Add(CorrelationVector.HeaderName, cVector.Value);
+                }
+
+                HttpResponseMessage resp = await Client.SendAsync(req);
 
```

Comment:
The code below this comment wasn't part of the change, but I noticed it and thought that I might have structured it differently by building the JsonResult in the conditional (add a then block to build the second) then after the conditional had a single return statement to return the JsonResult constructed within the conditional.  Just a thought....

Reply:
fixed